### DISCUSSION TITLE
Keycloak Custom Identity Provider Link

### DISCUSSION THREAD

**Discussion by bernhardriegler** [keycloak/keycloak#21743] [Q&A]
TL;DR: 

Is there a way to use a value from the JWT returned by the Identity Provider rather than the username?

Full story:

I have an OIDC Identity Provider and some lokal users in Keycloak.
Users should be able once authenticated to link their identity provider account to their current (local) account.
Keycloak does provide this functionality, however it will only use the username from the JWT ignoring all other fields in the token.

The service provider behind the identity provider will change the usernames on every login. The only real key to know which user is currently authenticated is to look for a specific field in the JWT.

I tried to add a mapper and also a username template to the identity provider configuration without any luck.

I found this issue: https://github.com/keycloak/keycloak/issues/11195
I am not sure if this not yet released change will address my issue.
I am looking forward for any hints and help.

I also looked through the keycloak core implementation when linking accounts and could not find a way to customize what is used for the link:
https://github.com/keycloak/keycloak/blob/22.0.0/services/src/main/java/org/keycloak/services/resources/IdentityBrokerService.java#L917

Thank you very much in advance!
